### DISCUSSION TITLE
Controlling multiple STEP400 on same network

### DISCUSSION THREAD

**Discussion by halpb** [ponoor/step-series-support#9] [Q&A] [answered]
Hello-

LOVE the STEP400, keep finding new ways to optimize my program and do things better.  Unfortunately, I've run out of motors/sensor inputs and would like to explore connecting multiple STEP400 via an Ethernet network and controlling them from a single device.  I'm not a programmer and have more-or-less taught myself how to use OSC/MAX.

Obviously, each STEP400 would need to be at a unique IP, however it is not clear to me how to set program to dynamically change who it communicates to.

I would imagine I need to have the individual commands send to "s toSTEP400" and then "s toSTEP400-2" or something like that.  But not sure how to setup the network initialization such that a single PC can send/receive from multiple STEP400 on the network.

Thanks!
![image](https://user-images.githubusercontent.com/93683200/151890397-f3a6ecbd-025e-449b-80b0-688e1b28e329.png)

**Comment by kanta** [comment, marked as answer]
Hello, thank you for using STEP400!

In Max, you can use multiple `udpsend` and `udpreceive` at the same time so you can patch them like this;

![image](https://user-images.githubusercontent.com/1097902/151894335-15009c21-3342-45c3-b4a3-ebe8b3649566.png)
I hope this answers your question. If I have misunderstood, please let me know!

**Reply by halpb** [reply]
Thanks Kanta!

I shamelessly used your reference Max patcher that had virtually all of the commands and have made it my own, practically invaluable to get started.

I will try this out, but seems straight forward enough.  Might splurge for a MAX800 as well.  

As an aside, one cool thing that you can do with the STEP400 boards is drive an electromagnet.  You simply set the mode to current mode and then alternate between /hardHiZ and /hardStop.  I also use the motor ports to create a very hacky enable line connecting a resistor to ground from one of the motor legs.  Need to be careful (trial and error) to make sure you aren't putting too much current through the resistor, but it works well to drive an enable to a external device.

Keep up the good work!

**Reply by kanta** [reply]
Your welcome!
I'm glad to see that STEP series can help Maxers, since I originally came from Max.

I've never heard to use these motor driver chips for electromagnets, what a brilliant idea! 
If you don't mind, could you show us a picture of the electromagnet in action in STEP400? I'm very curious.

**Comment by halpb** [comment]
Sure thing!  Here is a video driving a 12V electromagnet.  

https://user-images.githubusercontent.com/93683200/152219095-0b2b1885-fcf0-457d-bf4c-f8b26af338cd.mp4

**Reply by kanta** [reply]
Wow it's actually working! Thank you for sharing the fascinating video!
